INI: values with spaces can be single or double quoted

akorn opened the pull request:
Ideally the page should enumerate all characters whose presence requires the value to be quoted (I'm guessing that includes quotes and possibly backslashes?).

Also: maybe you can also escape whitespace with a backslash, not just quote it?

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
The page said "When declared inline with the host, INI values are interpreted as Python literal structures (strings, numbers, tuples, lists, dicts, booleans, None). Host lines accept multiple key=value parameters per line. Therefore they need a way to indicate that a space is part of a value rather than a separator." but failed to say what that way was. I added quoting as one possible way.
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Docs Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
inventory(?)

##### ADDITIONAL INFORMATION
None

akorn committed:
INI: values with spaces can be single or double quoted

Ideally the page should enumerate all characters whose presence requires the value to be quoted (I'm guessing that includes quotes and possibly backslashes?).

Also: maybe you can also escape whitespace with a backslash, not just quote it?

briantist reviewed (approved):
Looks good to me! 
@sivel provided the answer in IRC originally as well.

I am not sure about the questions raised by @akorn though, re: escaping on other characters.

samccann commented:
Ok we can leave it open for a few days to see if @sivel or someone can answer that final question. Thanks @akorn for the ansible docs fix and welcome to Ansible!

akorn commented:
> I think it's just sufficient to say that the value can be quoted. I don't think we should call out both single and double quotes. If we want, we can link to documentation for `shlex` here.
> 
> https://docs.python.org/3/library/shlex.html#parsing-rules

That's a good idea, a link should definitely be included.

> However, I don't think we should try to duplicate the info from the python docs into our docs. Which is why I think just saying "The value can be quoted" is sufficient.

I agree it's not necessary to duplicate the Python documentation, but I think it's useful to explicitly state what kinds of quotes will work. As a user, if I were reading that document and it just said "can be quoted" I would be annoyed because it didn't say how and I would need to find out by experimenting. "Single or double" isn't excessively wordy, isn't likely to change, and doesn't duplicate too much of the Python documentation.

samccann commented:
Hi @akorn - can you add the link and POSIX note to this PR?   As for whether to have just 'quoted' vs 'single or double quoted'... I  would lean toward @sivel's comment that saying quoted is enough. There is only single or double quoted options, right? But if you feel strongly that it helps the user to state it explicitly, I'm fine with that as well.

Thanks!

akorn committed:
Add link to Python shlex parsing rules

I also reduced the wordiness ("single quoted or double quoted" -> "quoted (single or double)"

akorn commented:
> Hi @akorn - can you add the link and POSIX note to this PR? As for whether to have just 'quoted' vs 'single or double quoted'... I would lean toward @sivel's comment that saying quoted is enough. There is only single or double quoted options, right? But if you 

Yes, there are only these two options, but either both work or only one of them works. I think adding three words to point out that both work is better than leaving the user guessing, or having to find out by trial and error, or having to check the shlex documentation.

I made it "quoted (single or double)", and added the shlex parsing rules link (correctly, I hope -- I'm not terribly good with rst).
